lib/block.ml

```diff
@@ -146,7 +146,8 @@ let pp_contents ?syntax ppf t =
         Fmt.pf ppf "%s" (String.concat "\n" t.contents)
       else Fmt.pf ppf "\n%a" (pp_lines syntax t) (List.map lstrip t.contents)
   | Some Cram | Some Normal | None ->
-      Fmt.pf ppf "%a\n" (pp_lines syntax t) t.contents
+      if List.length t.contents > 0 then
```

Comment:
How about replacing the `List.length` in this function by pattern matching. Since we only care about the list being a singleton or being non-empty we don't need to walk the entire list.

I'm thinking something like:
```ocaml
let pp_contents ?syntax ppf t =
  match syntax, t.contents with
  | Some Syntax.Mli, [ _ ] ->
      Fmt.pf ppf "%s" (String.concat "\n" t.contents)
  | Some Syntax.Mli, _ ->
      Fmt.pf ppf "\n%a" (pp_lines syntax t) (List.map lstrip t.contents)
  | (Some Cram | Some Normal | None), [] -> ()
  | (Some Cram | Some Normal | None), _ ->
      Fmt.pf ppf "%a\n" (pp_lines syntax t) t.contents
```

Reply:
It's better indeed, done.